src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -1406,7 +1406,27 @@ case class MatrixMapCols(child: MatrixIR, newCol: IR, newKey: Option[IndexedSeq[
     val colValuesType = TArray(prev.typ.colType)
     val vaType = prev.typ.rvRowType
 
-    val transformInitOp: (Int, IR) => IR = { (nAggs, initOpIR) =>
+    val (rvAggs,
+      (preInitOp, initOpCompiler),
+      (preSeqOp, seqOpCompiler),
+      aggResultType,
```

Comment:
Don't you have to change the other places that use CompileWithAggregators?

Reply:
I maintained the old interface and added one function to handle the MatrixMapCols case